src/color.cpp

```diff
@@ -6,756 +6,412 @@
- init_pair(10, COLOR_BLACK,   COLOR_GREEN  );
- init_pair(11, COLOR_BLACK,   COLOR_BLUE   );
- init_pair(12, COLOR_BLACK,   COLOR_CYAN   );
- init_pair(13, COLOR_BLACK,   COLOR_MAGENTA);
- init_pair(14, COLOR_BLACK,   COLOR_YELLOW );
-
- // Highlighted - blue background
- init_pair(15, COLOR_WHITE,   HILIGHT);
- init_pair(16, COLOR_RED,     HILIGHT);
- init_pair(17, COLOR_GREEN,   HILIGHT);
- init_pair(18, COLOR_BLUE,    HILIGHT);
- init_pair(19, COLOR_CYAN,    HILIGHT);
- init_pair(20, COLOR_BLACK,   HILIGHT);
- init_pair(21, COLOR_MAGENTA, HILIGHT);
- init_pair(22, COLOR_YELLOW,  HILIGHT);
-
- // Red background - for monsters on fire
- init_pair(23, COLOR_WHITE,   COLOR_RED);
- init_pair(24, COLOR_RED,     COLOR_RED);
- init_pair(25, COLOR_GREEN,   COLOR_RED);
- init_pair(26, COLOR_BLUE,    COLOR_RED);
- init_pair(27, COLOR_CYAN,    COLOR_RED);
- init_pair(28, COLOR_MAGENTA, COLOR_RED);
- init_pair(29, COLOR_YELLOW,  COLOR_RED);
-
- init_pair(30, COLOR_BLACK,   COLOR_BLACK);
- init_pair(31, COLOR_WHITE,   COLOR_BLACK);
-
- init_pair(32, COLOR_BLACK, COLOR_WHITE);
- init_pair(33, COLOR_WHITE, COLOR_WHITE);
- init_pair(34, COLOR_RED, COLOR_WHITE);
- init_pair(35, COLOR_GREEN, COLOR_WHITE);
- init_pair(36, COLOR_YELLOW, COLOR_WHITE);
- init_pair(37, COLOR_BLUE, COLOR_WHITE);
- init_pair(38, COLOR_MAGENTA, COLOR_WHITE);
- init_pair(39, COLOR_CYAN, COLOR_WHITE);
-
- init_pair(40, COLOR_BLACK, COLOR_GREEN);
- init_pair(41, COLOR_WHITE, COLOR_GREEN);
- init_pair(42, COLOR_RED, COLOR_GREEN);
- init_pair(43, COLOR_GREEN, COLOR_GREEN);
- init_pair(44, COLOR_YELLOW, COLOR_GREEN);
- init_pair(45, COLOR_BLUE, COLOR_GREEN);
- init_pair(46, COLOR_MAGENTA, COLOR_GREEN);
- init_pair(47, COLOR_CYAN, COLOR_GREEN);
-
- init_pair(48, COLOR_BLACK, COLOR_YELLOW);
- init_pair(49, COLOR_WHITE, COLOR_YELLOW);
- init_pair(50, COLOR_RED, COLOR_YELLOW);
- init_pair(51, COLOR_GREEN, COLOR_YELLOW);
- init_pair(52, COLOR_YELLOW, COLOR_YELLOW);
- init_pair(53, COLOR_BLUE, COLOR_YELLOW);
- init_pair(54, COLOR_MAGENTA, COLOR_YELLOW);
- init_pair(55, COLOR_CYAN, COLOR_YELLOW);
-
- init_pair(56, COLOR_BLACK, COLOR_MAGENTA);
- init_pair(57, COLOR_WHITE, COLOR_MAGENTA);
- init_pair(58, COLOR_RED, COLOR_MAGENTA);
- init_pair(59, COLOR_GREEN, COLOR_MAGENTA);
- init_pair(60, COLOR_YELLOW, COLOR_MAGENTA);
- init_pair(61, COLOR_BLUE, COLOR_MAGENTA);
- init_pair(62, COLOR_MAGENTA, COLOR_MAGENTA);
- init_pair(63, COLOR_CYAN, COLOR_MAGENTA);
-
- init_pair(64, COLOR_BLACK, COLOR_CYAN);
- init_pair(65, COLOR_WHITE, COLOR_CYAN);
- init_pair(66, COLOR_RED, COLOR_CYAN);
- init_pair(67, COLOR_GREEN, COLOR_CYAN);
- init_pair(68, COLOR_YELLOW, COLOR_CYAN);
- init_pair(69, COLOR_BLUE, COLOR_CYAN);
- init_pair(70, COLOR_MAGENTA, COLOR_CYAN);
- init_pair(71, COLOR_CYAN, COLOR_CYAN);
+    if ( mapColors[sName].first == 0 ) {
+        return 0;
+    }
 
+    return ( mapColors[sName].second != "" ) ? mapColors[mapColors[sName].second].first : mapColors[sName].first;
```

Comment:
The map look up `mapColors[sName]` is done 3 times, why not use an iterator and search the entry only once?
```C++
auto const iter = mapColors.find( sName );
if( iter == mapColors.end() ) {
    // not found -> return default or whatever
    return ...;
}
auto &entry = iter->second; // this is the same as mapColors[sName]
if( entry.first == 0 ) { return 0; }
....
```